lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -269,20 +271,35 @@ updateState
     -> Wallet s
 updateState s (Wallet u tip _ bp) = Wallet u tip s bp
 
--- | Apply Block is the primary way of making the wallet evolve. It returns the
--- updated wallet state, as well as a set of all transactions belonging to the
--- wallet discovered while applying the block.
+-- | Represents a set of data relevant to a particular wallet, discovered as
+--   the result of applying a block to that wallet.
+data ApplyBlockResult = ApplyBlockResult
+    { delegation :: !(Maybe PoolId)
+        -- ^ Any stake delegation on behalf of the wallet.
+    , transactions :: ![(Tx, TxMeta)]
+        -- ^ The set of transactions that affect the wallet.
+    } deriving (Generic, Show, Eq)
+
+-- | Apply a single block to a wallet.
+--
+-- This is the primary way of making a wallet evolve.
+--
+-- Returns an updated wallet, as well as the set of data relevant to the wallet
+-- that were discovered while applying the block.
+--
 applyBlock
-    :: Block
-    -> Wallet s
-    -> ([(Tx, TxMeta)], Wallet s)
-applyBlock !b (Wallet !u _ s bp) =
-    let
-        ((txs, u'), s') = prefilterBlock b u s
-    in
-        ( txs
-        , Wallet u' (b ^. #header) s' bp
-        )
+    :: Wallet s
+    -> Block
+    -> (ApplyBlockResult, Wallet s)
```

Comment:
**[suggestion ]** 

Change back to `Block -> Wallet s -> (ApplyBlockResult, Wallet s)`

**Why?**

As mentioned before, this is a rather common pattern in Haskell, `s -> (result, s)`. Having `s` as the last argument of the function makes it easier to re-use functions out of the box like [state](https://hackage.haskell.org/package/transformers-0.5.6.2/docs/Control-Monad-Trans-State-Lazy.html#v:state). 

See also:

- [genChange](https://input-output-hk.github.io/cardano-wallet/haddock/edge/cardano-wallet-core-2019.11.18/Cardano-Wallet-Primitive-AddressDiscovery.html#v:genChange)
- [isOurs](https://input-output-hk.github.io/cardano-wallet/haddock/edge/cardano-wallet-core-2019.11.18/Cardano-Wallet-Primitive-AddressDiscovery.html#v:isOurs)

Reply:
> Change back to `Block -> Wallet s -> (ApplyBlockResult, Wallet s)`

Done.